src/main/java/de/tum/in/www1/artemis/service/QuizExerciseImportService.java

```diff
@@ -97,6 +100,8 @@ private void copyQuizQuestions(QuizExercise importedExercise, QuizExercise newEx
                 }
             }
             else if (quizQuestion instanceof DragAndDropQuestion dndQuestion) {
+                // Check whether dndQuestion.getBackgroundFilePath() is actually a background file path
+                QuizExerciseImportService.checkIfBackgroundFilePathIsABackgroundFilePathElseThrow(dndQuestion.getBackgroundFilePath());
```

Comment:
There should also be a check for `dragItem.setPictureFilePath(fileService.copyExistingFileToTarget(dragItem.getPictureFilePath(), FilePathService.getDragItemFilePath(), null));` below, otherwise you can simply use the drag item pictures instead of the background images to circumvent the check.

Reply:
Introduced `FileService.sanitizeByCheckingIfPathContainsSubPathElseThrow(URI.create(dragItem.getPictureFilePath()), URI.create("/api/" + FileService.DRAG_AND_DROP_PICTURE_SUBPATH + "/"));` in [69a0c8b408b16a8e68ace79dfa4cee7927e48db0](https://github.com/ls1intum/Artemis/pull/7170/commits/69a0c8b408b16a8e68ace79dfa4cee7927e48db0).